dom.go

```diff
@@ -259,7 +259,7 @@ func (h *HTML) reconcile(prev *HTML) {
 
 	// TODO better list element reuse
 	for i, nextChild := range h.children {
-		if i >= len(prev.children) {
+		if i >= len(prev.children) || h.new {
```

Comment:
Could you add a simple `// TODO: write tests for h.new usage here` comment, just so I don't forget?